from modules.connection import *
from modules.vm import *
from modules.lun import *
from modules.ip import *
from modules.utils import *
import pandas as pd
import ovirtsdk4
import sys

try:
	connection = Connection()
	conn = connection.connection()
	conn.test(raise_exception = True)  # return true if connect successful
	utils = Utils(conn)
except ovirtsdk4.AuthError :
	print('Authentication failed, please check adminrc')
	sys.exit()

path = '/home/centos/rhvm/test_data.xlsx'
data = pd.read_excel(path)
print(data.shape)


luns = []
luns.append(LUN('4c281778-2e65-4679-a751-845386b8054e',True))

ips = []
ips.append(IP('10.1.21.22', 'nic1'))

vm = {}
vm['name'] = 'centos'
vm['ips'] = ips 
vm['vcpu'] = 4
vm['ram'] = 4 
vm['luns'] = luns
vm['cluster'] = 'managed' 
myvm = VM(vm)
utils.create_vm(myvm)
conn.close()

